src/gossip/service.zig

```diff
@@ -828,20 +823,18 @@ pub const GossipService = struct {
                 const prng_seed: u64 = @intCast(std.time.milliTimestamp());
                 var prng = std.Random.Xoshiro256.init(prng_seed);
                 try self.rotateActiveSet(prng.random());
-                last_push_ts = getWallclockMs();
             }
 
             // publish metrics
-            const stats_publish_elapsed_ts = getWallclockMs() - last_stats_publish_ts;
-            if (stats_publish_elapsed_ts > PUB_GOSSIP_STATS_INTERVAL_MS) {
+            if (stats_publish_timer.read().as_millis() > PUB_GOSSIP_STATS_INTERVAL_MS) {
+                defer stats_publish_timer.reset();
                 try self.collectGossipTableMetrics();
-                last_stats_publish_ts = getWallclockMs();
             }
 
             // sleep
-            const elapsed_ts = getWallclockMs() - top_of_loop_ts;
-            if (elapsed_ts < GOSSIP_SLEEP_MILLIS) {
-                const time_left_ms = GOSSIP_SLEEP_MILLIS - elapsed_ts;
+            if (loop_timer.read().as_millis() < GOSSIP_SLEEP_MILLIS) {
+                defer loop_timer.reset();
+                const time_left_ms = GOSSIP_SLEEP_MILLIS -| loop_timer.read().as_millis();
```

Comment:
We could probably start replacing all of this stuff with `Duration` at some point. Doesn't need to be in this PR, just thinking out loud.

Reply:
created a ticket - https://github.com/orgs/Syndica/projects/2/views/1?pane=issue&itemId=71010101